### PR TITLE
Couple of fixes

### PR DESCRIPTION
**Pull request by karthiknadig**
1. Markdown shows progress during publish.
2. Removed credential dependency from Monitor.

### CODE REVIEW

File: src/Package/Impl/Publishing/Commands/PreviewCommand.cs [hunk 1]
```diff
@@ -81,14 +83,15 @@ internal abstract class PreviewCommand : ViewCommand {
             _lastCommandTask = Task.Run(async () => {
                 // Get list of installed packages and verify that all the required ones are installed
                 var packages = await workflow.Packages.GetInstalledPackagesAsync();
-                // Text buffer operations should be performed in UI thread
-                await _coreShell.SwitchToMainThreadAsync();
                 if (packages.Any(p => p.Package.EqualsIgnoreCase(flavorHandler.RequiredPackageName)) && CheckPrerequisites()) {
+                    // Text buffer operations should be performed in UI thread
```
Comment (MikhailArkhipov) [inline src/Package/Impl/Publishing/Commands/PreviewCommand.cs]: Merge:

```
// Text buffer operations should be performed in UI thread		
 -                await _coreShell.SwitchToMainThreadAsync();
```

should be before `CheckPrerequisites`

File: src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs [hunk 2]
```diff
@@ -30,7 +33,9 @@ internal sealed class RmdPublishHandler : IMarkdownFlavorPublishHandler {
 
         public async Task PublishAsync(IRSession session, ICoreShell coreShell,  IFileSystem fs, string inputFilePath, string outputFilePath, PublishFormat publishFormat, Encoding encoding) {
             try {
-                await RMarkdownRenderAsync(session, fs, inputFilePath, outputFilePath, GetDocumentTypeString(publishFormat), encoding.CodePage);
+                IApplicationShell appShell = coreShell as IApplicationShell;
+                IVsStatusbar statusBar = appShell?.GetGlobalService<IVsStatusbar>(typeof(SVsStatusbar));
```
Comment (MikhailArkhipov) [inline src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs]: Must be done in UI thread

File: src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs [hunk 3]
```diff
@@ -30,7 +33,9 @@ internal sealed class RmdPublishHandler : IMarkdownFlavorPublishHandler {
...
+                await RMarkdownRenderAsync(session, fs, inputFilePath, outputFilePath, GetDocumentTypeString(publishFormat), encoding.CodePage, statusBar);
             } catch (IOException ex) {
                 coreShell.ShowErrorMessage(ex.Message);
```
Comment (MikhailArkhipov) [inline src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs]: change please to `ShowErrorMessageAsync` so UI thread is guaranteed

File: src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs [hunk 4]
```diff
@@ -39,11 +44,25 @@ internal sealed class RmdPublishHandler : IMarkdownFlavorPublishHandler {
...
         }
 
-        private async Task RMarkdownRenderAsync(IRSession session, IFileSystem fs, string inputFilePath, string outputFilePath, string format, int codePage) {
+        private async Task RMarkdownRenderAsync(IRSession session, IFileSystem fs, string inputFilePath, string outputFilePath, string format, int codePage, IVsStatusbar statusBar) {
             using (var fts = new DataTransferSession(session, fs)) {
-                var rmd = await fts.SendFileAsync(inputFilePath);
-                var publishResult = await session.EvaluateAsync<ulong>($"rtvs:::rmarkdown_publish(blob_id = {rmd.Id}, output_format = {format.ToRStringLiteral()}, encoding = 'cp{codePage}')", REvaluationKind.Normal);
-                await fts.FetchFileAsync(new RBlobInfo(publishResult), outputFilePath);
+                string currentStatusText;
+                statusBar.GetText(out currentStatusText);
```
Comment (int19h) [inline src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs]: Doesn't IVs* use COM STA marshaling everywhere? I recall it was even deliberately done for some services that switched to managed code, so as to not break that contract.

Comment (MikhailArkhipov) [inline src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs]: We can try. Some do work :-)

**Comment by karthiknadig** [comment]
Ping

File: src/Package/Impl/Publishing/Commands/PreviewCommand.cs [hunk 5]
```diff
@@ -96,16 +96,16 @@ internal abstract class PreviewCommand : ViewCommand {
                             try {
                                 _fs.DeleteFile(_outputFilePath);
                             } catch (IOException ex) {
-                                CoreShell.ShowErrorMessage(ex.Message);
+                                AppShell.ShowErrorMessage(ex.Message);
```
Comment (MikhailArkhipov) [inline src/Package/Impl/Publishing/Commands/PreviewCommand.cs]: Changed quite a bit, please merge master

**Comment by MikhailArkhipov** [comment]
👍 pending master merge

**Comment by karthiknadig** [comment]
It is merged with master.

**Pull request merged**